### PR TITLE
[MIG][11.0] mail_queue_job

### PR DESCRIPTION
**Pull request by acsonefho**
Old module name: mail_connector_queue (from connector repo)

- [x] Move module
- [x] XML migration
- [x] Python migration
- [x] Add tests
- [x] Flake/Pylint
- [x] Setup

### CODE REVIEW

File: oca_dependencies.txt
Comment (sbidoul) [inline oca_dependencies.txt]: why?

Comment (acsonefho) [reply oca_dependencies.txt]: component and component_event modules comes from connector repo

Comment (sbidoul) [reply oca_dependencies.txt]: ah yes, that makes a nice circular dependency loop between connector and queue. Not sure if it is an issue in practice.

Maybe it's lighter to overwrite create/write manually for this simple use case? What do you think @lmignon @guewen ?

Comment (lmignon) [reply oca_dependencies.txt]: Indeed, It could be lighter to only overwrite 'create/write' methods into the mail.mail to avoid this dependency. On the other hand the code looks clean and is easy to read when using the component.event addon to add a hook on the create/write method. The more I use components, the more I'm convinced that we should remove the dependency on connector to make it more generic. But it's an other debate.  Since the dependency is only required for this unique use into the addon I'm in favour to remove-it.

File: mail_queue_job/README.rst
Comment (guewen) [inline mail_queue_job/README.rst:2]: `Mail Queue Job`?

**Review by guewen — APPROVED** [review]
Great! Thanks :)

File: mail_queue_job/README.rst
Comment (acsonefho) [reply mail_queue_job/README.rst:2]: This file should be re-generated automatically after the merge?
And the name is automatically set by the name of the directory I suppose.

Comment (guewen) [reply mail_queue_job/README.rst:2]: I was not sure from where the name comes from, it's from the manifest file: https://github.com/OCA/maintainer-tools/blob/80f894b594810200386f176e19cd79bf4d7be8c5/tools/gen_addon_readme.template#L11

So indeed this is going to be correct after the automatic update as the name in the manifest is good.

**Review by jabibi — APPROVED** [review]
Code looks good to me. Thanks :+1:

File: mail_queue_job/models/mail_mail.py
Comment (lmignon) [inline mail_queue_job/models/mail_mail.py]: @acsonefho It would be nice to forward port https://github.com/OCA/queue/pull/66 in 11.0 To be able to use the new concept of Idendity key on queue job. This forward port could solve the problem of multiple jobs for the same email in a more 'elegant' way.

Comment (acsonefho) [reply mail_queue_job/models/mail_mail.py]: @lmignon It's done, thank you for your advice :)

File: mail_queue_job/models/mail_mail.py
Comment (guewen) [inline mail_queue_job/models/mail_mail.py:42]: I'm not sure what the intention here but beware of this: it won't prevent a concurrent transaction to create a job for the same email at the same time if the concurrent transaction happens within the same timeframe. It will however prevent the creation of a job at a different moment (only before the job is done) or within the same transaction. Is it enough for this use case?

Comment (acsonefho) [reply mail_queue_job/models/mail_mail.py:42]: The intention is to avoid sending the same `mail.mail` more than once (if a job is already planned for a `mail.mail` recordset, it shouldn't have another job who'll re-send the same `mail.mail`).
But when the job is finished, a user can try to re-sent the same `mail.mail` so it could create a new job for that.
About your first case (same timeframe and concurrent transaction) it should never happens because jobs are created into `create`/`write` so it's the same transaction.
Thank you for your review @guewen

**Pull request merged**
